hyperspy/model.py

```diff
@@ -924,475 +956,698 @@ def red_chisq(self):
+                * "custom" allows passing your own minimisation function as a
+                  keyword argument ``min_function``, with an optional
+                  gradient argument ``min_function_grad``. Not available
+                  if ``fitter`` is ``"leastsq"`` or ``"odr"``.
+
+        grad : bool or ["2-point", "3-point", "cs"] or callable, default False
+            If True, the analytical gradient is used (if defined) to
+            speed up the optimization. The keywords ``['2-point', '3-point', 'cs']``
+            can be used to select a finite difference scheme for numerical
+            estimation of the gradient with a relative step size. If it is a
+            callable, it should be a function that returns the gradient vector.
+            See :py:func:`scipy.optimize.minimize` for more details.
+        bounded : bool, default False
+            If True, performs bounded parameter optimization if
+            supported by ``fitter``.
+        ext_bounding : bool, default False
             If True, enforce bounding by keeping the value of the
             parameters constant out of the defined bounding area.
-        **kwargs : key word arguments
-            Any extra key word argument will be passed to the chosen
-            fitter. For more information read the docstring of the optimizer
-            of your choice in `scipy.optimize`.
+        update_plot : bool, default False
+            If True, the plot is updated during the optimization
+            process. It slows down the optimization, but it enables
+            visualization of the optimization progress.
+        weights : {None, "inverse_variance", np.ndarray}, default "inverse_variance"
+            If None:
+                No weighting is applied to the data.
+            If "inverse_variance":
+                If the attribute ``metada.Signal.Noise_properties.variance``
+                is defined as a ``Signal`` instance of the same
+                ``navigation_dimension`` as the signal, and ``method="ls"``,
+                then weighted least-squares fit is performed, using
+                the inverse of the noise variance as the weights.
+            If np.ndarray:
+                If the array has the same dimensions as the signal,
+                and ``method="ls"``, then weighted least squares
+                is performed using the array values as the weights.
+        print_info : bool, default False
+            If True, print information about the fitting results.
+        **kwargs : keyword arguments
+            Any extra keyword argument will be passed to the chosen
+            fitter. For more information, read the docstring of the
+            optimizer of your choice in :py:mod:`scipy.optimize`.
+
+        Returns
+        -------
+        None
+
+        Notes
+        -----
+        The chi-squared and reduced chi-squared statistics, and the
+        degrees of freedom, are computed automatically when fitting.
```

Comment:
Maybe it'll be good to add here that they are computed only when fitting using least-squares.

Reply:
Yes good idea - done.